### PR TITLE
Remove function invokation parens from documentation links.

### PR DESCRIPTION
**Pull request by frewsxcv**
This was never established as a convention we should follow in the 'More
API Documentation Conventions' RFC:

https://github.com/rust-lang/rfcs/blob/master/text/1574-more-api-documentation-conventions.md

### CODE REVIEW

**Comment by frewsxcv** [comment]
This PR was generated via:

```
find . -name "*.md" | xargs gsed -i 's/\[\([^ ]*\)()`\]/[\1`]/g'
find . -name "*.rs" | xargs gsed -i 's/\[\([^ ]*\)()`\]/[\1`]/g'
```

**Comment by frewsxcv** [comment]
cc @rust-lang/docs

**Comment by GuillaumeGomez** [comment]
I'm still sad about this. :'(

@bors: r+ rollup

File: src/libstd/thread/mod.rs [hunk 1]
```diff
@@ -90,29 +90,29 @@
...
+//! The [`thread::current`] function is available even for threads not spawned
 //! by the APIs of this module.
 //!
 //! ## Blocking support: park and unpark
 //!
 //! Every thread is equipped with some basic low-level blocking support, via the
-//! [`thread::park()`][`park()`] function and [`thread::Thread::unpark()`][`unpark()`]
-//! method. [`park()`] blocks the current thread, which can then be resumed from
-//! another thread by calling the [`unpark()`] method on the blocked thread's handle.
+//! [`thread::park()`][`park`] function and [`thread::Thread::unpark()`][`unpark`]
```
Comment (ollie27) [inline src/libstd/thread/mod.rs]: missed a couple

File: src/libstd/thread/mod.rs [hunk 2]
```diff
@@ -90,29 +90,29 @@
...
+//! method. [`park`] blocks the current thread, which can then be resumed from
+//! another thread by calling the [`unpark`] method on the blocked thread's handle.
 //!
 //! Conceptually, each [`Thread`] handle has an associated token, which is
 //! initially not present:
 //!
-//! * The [`thread::park()`][`park()`] function blocks the current thread unless or until
+//! * The [`thread::park()`][`park`] function blocks the current thread unless or until
```
Comment (ollie27) [inline src/libstd/thread/mod.rs]: and here

**Comment by steveklabnik** [comment]
@bors: r-

thanks @ollie27 

me too @GuillaumeGomez

**Comment by GuillaumeGomez** [comment]
Seems fixed now.

@bors: r+ rollup

**Pull request merged**